main/portable-scandir.c

```diff
@@ -0,0 +1,243 @@
+#include "general.h"
+/*
+ * Taken from https://raw.githubusercontent.com/ClusterLabs/pacemaker/master/replace/scandir.c
+ */
+
+/* scandir: Scan a directory, collecting all (selected) items into a an array.
+ *
+ * This code borrowed from 'libit', which can be found here:
+ *
+ * http://www.iro.umontreal.ca/~pinard/libit/dist/scandir/
+ *
+ * The original author put this code in the public domain.
+ * It has been modified slightly to get rid of warnings, etc.
+ *
+ * Below is the email I received from pinard@iro.umontreal.ca (Fran�ois Pinard)
```

Comment:
Encoding seems broken.
s/Fran�ois/François/
The original file was latin 1 (ISO 8859-1), but maybe it's batter to use UTF-8?